hpx/lcos/detail/async_implementations.hpp

```diff
@@ -155,16 +155,9 @@ namespace hpx { namespace detail
     };
 
     template <typename Action>
-    bool can_invoke_locally()
+    HPX_CONSTEXPR bool can_invoke_locally()
     {
-        return !traits::action_decorate_function<Action>::value &&
-            this_thread::get_priority() ==
-                static_cast<threads::thread_priority>(
-                    traits::action_priority<Action>::value) &&
-            this_thread::get_stack_size() ==
-                threads::get_stack_size(
-                    static_cast<threads::thread_stacksize>(
-                        traits::action_stacksize<Action>::value));
```

Comment:
I suggest to rather change it to
```
return !traits::action_decorate_function<Action>::value &&
    this_thread::get_stack_size() >=
        threads::get_stack_size(		
            static_cast<threads::thread_stacksize>(		
                traits::action_stacksize<Action>::value));
```
That way, we don't run into the problem if an action should run on a larger stack than what we have currently.
The priority change is correct though, as we intend to run the action immediately anyway.

Reply:
The problem with not checking the priority is that the following test fails then:
http://cdash.cscs.ch/testDetails.php?test=11627793&build=73031
For the stacksize, we get this failure:
http://cdash.cscs.ch/testDetails.php?test=11628196&build=73031

Reply:
Checking the priority does not make any sense as the function will be executed right away. I'd suggest correcting the test. I agree on the stack size change you suggested, though. That should also make the test pass.

Reply:
The tests don't use direct actions. The actions are just executed locally.

Reply:
Sure, I didn't say anything about direct actions, though. I still think that the tests are overly constrained.

Reply:
@sithhell Your comments have been addressed.